src/Lib/Modes/PathEditMode.vala

```diff
@@ -145,12 +142,117 @@ public class Akira.Lib.Modes.PathEditMode : AbstractInteractionMode {
     }
 
     public override bool button_press_event (Gdk.EventButton event) {
-        if (!is_edit_path) {
+        // If button is clicked in edit mode, check if some point was clicked.
+        if (is_edit_path) {
+            int[] index = new int[3];
+            index[0] = index[1] = index[2] = -1;
+
+            var sel_type = edit_model.hit_test (event.x, event.y, ref index);
```

Comment:
@mbfraga Don't actually need to do this. Hit test on zoom works properly.
This is because the event is scaled and then propagated from the **ViewCanvas** itself.

Reply:
Zooming in past a certain level completely breaks editing--so I think something is still broken. Even so...we really need to get rid of scaling the event since that's bad behavior. i'll probably do that sooner rather than later.